1주차 과제 1, let을 제거해보자

xunxee opened the pull request:
- 재할당을 하지 않으면서 기존과 동일한 결과를 만들기 위해서
  함수의 매개변수를 사용하여 문제를 해결해나가고자 생각했습니다.
- 개발 환경을 구축하고 강의 내용을 이해하는데 많은 시간이 필요했습니다.
  문제를 해결하지 못했지만 commit을 남겨봅니다.

xunxee committed:
1주차 과제 1, let을 제거해보자

- 재할당을 하지 않으면서 기존과 동일한 결과를 만들기 위해서
  함수의 매개변수를 사용하여 문제를 해결해나가고자 생각했습니다.
- 개발 환경을 구축하고 강의 내용을 이해하는데 많은 시간이 필요했습니다.
  문제를 해결하지 못했지만 commit을 남겨봅니다.

daadaadaah commented on src/index.js:
```diff
@@ -1,2 +1,55 @@
+          )
+        </button>
+      </p>
+      {[1, 2, 3].map((num) => (
```
축약어보다 number 이렇게 모두 기재해주는 습관을 가져봅시다!ㅎㅎ

daadaadaah commented on src/index.js:
```diff
@@ -1,2 +1,55 @@
+    render();
+  }
+
+  function handleClickNumber(value) {
```
코드 수정하실 때, `value` 같은 추상적인 네이밍보다 좀더 의도가 드러나는 네이밍이 없을까? 한번 고민해보세요~ㅎㅎ

daadaadaah reviewed (commented):
과제가 다 완성되지 않았지만, 용기있게 PR 날려주신 점 너무 좋습니다!👍
계속 진행해주세요🚀

xunxee committed:
리뷰 반영 및 함수의 매개변수(count)를 사용하여 접근

1. 리뷰를 반영했습니다.
  - 축약어(num) -> 'number'으로 수정했습니다.
  - handleClickNumber(value)와 같은 추상적 네이밍을
    handleClickNumber(buttonNumber)으로 수정했습니다.
2. function render의 매개변수 count에 값을 할당했습니다.
  - render 함수의 중첩 함수로 handleClick 함수와
    handleClickNumber 함수를 넣어주어서 render 함수의 매개변수에 접근하도록 만들었습니다.

daadaadaah commented on src/index.js:
```diff
@@ -19,20 +20,18 @@ function createElement(tagName, props, ...children) {
 }
 
-function render(count = 0) {
+function render(count = 1) {
```
기본값을 0 -> 1 로 바꿔 주신 특별한 이유가 있으신가용?ㅎㅎ

xunxee replied:
앗, 제가 실수로 0에서 1로 바꾸었네요!
다시 0으로 수정했습니당 ㅎㅎ

daadaadaah commented on src/index.js:
```diff
@@ -41,9 +40,9 @@ function render(count = 0) {
-      {[1, 2, 3].map((num) => (
-        <button type="button" onClick={() => handleClickNumber}>
-          {num}
+      {[1, 2, 3].map((buttonNumber) => (
```
FullName으로 기재해주신 점 너무 좋습니다!
그런데, `number`로 해도 의미상 문제가 없을 것 같은데, 어떻게 생각하시나요?ㅎㅎ

xunxee replied:
number로 해도 충분히 의미가 전달되는 것 같습니당!
감사합니다 🐥

daadaadaah reviewed (changes requested):
commit 안에 어떤 작업을 했는지 기재해주신 점 너무 좋습니다! 👍
좀더 작업 내역을 잘 관리하고 싶으시면, 1개의 commit에 모든 작업이 들어가는게 아니라, 작업별로 commit 남겨보세요~ㅎㅎ

xunxee committed:
2번째 리뷰 반영

1. function render의 매개변수 count의 기본 값을 1에서
  다시 0으로 수정했습니다.
2. handleClickNumber 함수의 매개변수명을 buttonNumber에서
  number로 수정했습니다.

daadaadaah commented on src/index.js:
```diff
@@ -1,2 +1,54 @@
 /* eslint-disable react/react-in-jsx-scope, react/jsx-filename-extension */
+/* eslint no-use-before-define: ["error", { "functions": false }] */
```
이 주석 이제 삭제해도 괜찮지 않을까요? ㅎㅎ
